### ISSUE TITLE
TODOs display incorrectly inside references

### ISSUE THREAD

**Issue by tmladek** [dendronhq/dendron#339]
**Describe the bug**
If I reference a note which uses the `- [ ]` syntax to denote checkboxes / TODOs, it does not render correctly inside a reference.

**To Reproduce**
1. Create a file, add some `- [ ] TODOs`
1. Create another file, use a reference `((ref:[[other.file]]))`

**Expected behavior**
The `- [ ]` syntax to render as checkboxes, as in the original document.

**Screenshots**
Bug:
![image](https://user-images.githubusercontent.com/5217341/99066386-2c6d6680-25a9-11eb-952b-d3f52b9b714a.png)

The way it looks in the original file:
![image](https://user-images.githubusercontent.com/5217341/99066509-60488c00-25a9-11eb-9abe-f5f35fb535c2.png)

**Desktop (please complete the following information):**
VS Code version:
```
Version: 1.51.1
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-10T23:31:29.624Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.9.6-arch1-1
```
Dendron version: 0.15.0

**Dendron Log file**
I don't think they should be necessary, will of course provide otherwise.

**Comment by kevinslin** [comment]
Thanks for the issue. Not a straightforward fix because the markdown preview has its own dynamic todo rendering that we bypass when we do a note reference. 

We are planning on replacing the markdown preview with our own preview around end of January. If this isn't an urgent issue, I would recommend waiting for that which would fix this and some other inconsistencies with the preview

**Comment by tmladek** [comment]
It really isn't an urgent issue - after all, Markdown is good at remaining readable/functional even as plain text :)

Thanks for the heads up, looking forward to the new preview!

**Comment by kpathakota** [comment]
This should get resolved in #643 feature

**Comment by aleksey-rowan** [comment]
@tmladek, checkboxes are now rendered correctly with the V2 Preview:

![image](https://user-images.githubusercontent.com/79934725/152913276-18aabe8b-cd1d-447b-8732-4311156e9b28.png)
